x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/sequence/OrdinalGroup.java

```diff
@@ -52,11 +52,27 @@ void add(E element) {
      * The element and everything before it is removed.
      */
     E trimBefore(Ordinal ordinal) {
+        return trimBefore(ordinal, true);
+    }
+
+    /**
+     * Returns the latest element from the group that has its timestamp
+     * less than the given argument alongside its position in the list.
+     * Everything before the element it is removed. The element is kept.
```

Comment:
```suggestion
     * Everything before this element is removed. The element is kept.
```

Reply:
Will pick them up in a separate PR to avoid another build.